utils/ghc-pkg/Main.hs

```diff
@@ -593,6 +597,23 @@ readFromSettingsFile settingsFile f = do
       Right archOS -> Right archOS
       Left e -> Left e
 
+getTarget :: [Flag] -> IO (Maybe String)
```

Comment:
I wonder if we want to have a comment here that explains target and that we either pick it from the arguments or from the filename; and that argument has precedent? Ideally with 2-3 example invocations?